bolt-modules/prompt/lib/puppet/functions/prompt.rb

```diff
@@ -30,14 +34,20 @@ def prompt(prompt, options = {})
                               action: 'prompt')
     end
 
-    options = options.transform_keys(&:to_sym)
-
+    options  = options.transform_keys(&:to_sym)
     executor = Puppet.lookup(:bolt_executor)
+
     # Send analytics report
     executor.report_function_call(self.class.name)
 
+    # Require default to be a string value
+    if options.key?(:default) && !options[:default].is_a?(String)
+      raise Bolt::ValidationError, "Option 'default' must be a string"
+    end
```

Comment:
Should this allow `nil`/`Undef` as a default?

Reply:
Hmm I'd be interested in waiting until someone asks for it to understand their use case better - I think an empty string will work for basically making the prompt "optional". It'd be useful to include that use case as an example I think:
```
$prompt = prompt("Password please", "default" => "")
if $prompt.empty {
  # Fail the plan with a custom message
  fail_plan("Please provide a password to the prompt")
} elsif {
# Some validation here...
} else {
  run_script(...)
}
```
Something along those lines.